packages/slice-machine/components/ScreenshotChangesModal/index.tsx

```diff
@@ -0,0 +1,205 @@
+import SliceMachineModal from "@components/SliceMachineModal";
+
+import { useSelector } from "react-redux";
+import { isModalOpen } from "@src/modules/modal";
+import { SliceMachineStoreType } from "@src/redux/type";
+import { ModalKeysEnum } from "@src/modules/modal/types";
+
+import { Flex, Heading, Close, Box, Text } from "theme-ui";
+
+import { ComponentUI } from "@lib/models/common/ComponentUI";
+import { useState } from "react";
+
+const VariationIcon: React.FC<{ isValid?: boolean }> = ({ isValid }) => (
+  <Flex
+    sx={{
+      p: 2,
+      border: "1px solid #E4E2E4",
+      borderRadius: "6px",
+      bg: "#FFF",
+    }}
+  >
+    {isValid ? (
+      <AiOutlinePicture style={{ fontSize: "16px" }} />
+    ) : (
+      <RiErrorWarningLine style={{ fontSize: "16px", color: "#ED811C" }} />
+    )}
+  </Flex>
+);
+
+const VariationsList = ({
+  slice,
+  onSelectVariation,
+  selectedSliceVariation,
+}: {
+  slice: ComponentUI;
+  isSelected?: boolean;
+  selectedSliceVariation: [string, string];
+  onSelectVariation: (sliceID: string, variationID: string) => void;
+}) => (
+  <>
+    <Heading
+      as="h4"
+      sx={{
+        whiteSpace: "nowrap",
+        overflow: "hidden",
+        textOverflow: "ellipsis",
+      }}
+    >
+      {slice.model.name}
+    </Heading>
+    <Box>
+      <Box
+        as="ul"
+        sx={{
+          pl: 0,
+          my: 3,
+        }}
+      >
+        {slice.model.variations.map((variation) => {
+          const isSelectedVariation =
+            selectedSliceVariation[0] === slice.model.id &&
+            selectedSliceVariation[1] === variation.id;
+          return (
+            <Box
+              as="li"
+              sx={{
+                listStyle: "none",
+                my: 2,
+                p: 2,
+                cursor: "pointer",
+                borderRadius: "4px",
+                ...(isSelectedVariation && {
+                  bg: "#EDECEE",
+                  fontWeight: "500",
+                }),
+                ":hover": {
+                  bg: "#EDECEE",
+                },
+              }}
+              onClick={() => onSelectVariation(slice.model.id, variation.id)}
+            >
+              <Flex
+                sx={{
+                  alignItems: "center",
+                }}
+              >
+                <VariationIcon isValid={!!slice.screenshots[variation.id]} />
+                <Text sx={{ ml: 2 }}>{variation.name}</Text>
+              </Flex>
+            </Box>
+          );
+        })}
+      </Box>
+    </Box>
+  </>
+);
+
+const ScreenshotChangesModal = ({
+  slices,
+  defaultSelectedSliceVariation = [],
+  onClose,
+}: {
+  slices: ComponentUI[];
+  defaultSelectedSliceVariation?: [string, string] | [];
+  onClose: () => void;
+}) => {
+  const { isOpen } = useSelector((store: SliceMachineStoreType) => ({
+    isOpen: isModalOpen(store, ModalKeysEnum.SCREENSHOTS),
+  }));
+  const [defaultSelectedSliceID, defaultSelectedSliceVariationID] =
```

Comment:
should be an object, `{ sliceID, variationID }` makes it clearer and more secure than an array.

Reply:
It should also be a type created for this, i see it's being used in a useState below.

Reply:
I agree!

Reply:
updated